zebrad/src/commands/entry_point.rs

```diff
@@ -63,28 +61,22 @@ impl EntryPoint {
         "start"
     }
 
+    /// Checks if the provided arguments include a subcommand
+    fn should_add_default_subcommand(&self) -> bool {
+        self.cmd.is_none()
+    }
+
     /// Process command arguments and insert the default subcommand
     /// if no subcommand is provided.
     pub fn process_cli_args(mut args: Vec<OsString>) -> clap::error::Result<Vec<OsString>> {
-        // Check if the provided arguments include a subcommand
-        let should_add_default_subcommand = EntryPoint::try_parse_from(&args)?.cmd.is_none();
+        let entry_point = EntryPoint::try_parse_from(&args)?;
 
         // Add the default subcommand to args after the top-level args if cmd is None
-        if should_add_default_subcommand {
-            // try_parse_from currently produces an error if the first argument is not the binary name,
-            let mut num_top_level_args = 1;
-
-            // update last_top_level_arg_idx to the number of top-level args
-            for (idx, arg) in args.iter().enumerate() {
-                num_top_level_args = match arg.to_str() {
-                    Some("--verbose" | "-v" | "--version" | "-V" | "--help") => idx + 1,
-                    Some("--config" | "-c") => idx + 2,
-                    _ => num_top_level_args,
-                }
+        if entry_point.should_add_default_subcommand() {
+            args.push(EntryPoint::default_cmd_as_str().into());
+            for filter in entry_point.filters {
+                args.push(filter.into())
```

Comment:
It duplicates the top-level filters args, but will parse and ignores the first set.